_posts/2017-11-07-amp-roadshow.md

```diff
@@ -0,0 +1,114 @@
+
+AMP re-implements many DOM elements we know from HTML, that interact with "slow" external dependencies, such as loading images (and rendering them). AMP focuses on optimizing the user experience, which is rendering speed to name the most important one. For example in AMP you use `<amp-img>` instead of `<img>`. The custom elements `<amp-youtube>`, `<amp-video>` and `<amp-twitter>` make it very obvious what AMP wants to control: the loading pipeline of anything that might make the rendering slow. There are various extensions though, so there might be many even more customized elements that can be used with AMP.
+
+While the reasoning behind it and the techniques used make a lot of sense, AMP also limits down the possibility of what one can use. Because if you want your site to be a valid AMP site you need to run it against the [AMP validator][amp-validator], which checks the site to fulfill certain criteria, for example that all CSS must be inlined and must not be exceed 50kB.
+
+[web components]: https://www.webcomponents.org/
+[tweet-1]: https://twitter.com/wolframkriesing/status/927825238766882818
+[tweet-2]: https://twitter.com/wolframkriesing/status/927835789404319744
+[amp-validator]: https://validator.ampproject.org/
+
+## AMP in Production
+
+<div style="float: left; padding: 1rem;">
+<img src="{{site.baseurl}}/img/posts/2017-11-amp-roadshow/3-in-prod.jpg" alt="amp in production" width="100" class="sizeup-onhover-image scale3 origin-left-top" />
+<br/><em>AMP in production</em>
+</div>
+
+After the basics, the "why" and "how" of AMP, [Rudy][rudy] (???????? was it Rudy?????????) went to show us about how to get AMP live. Three basic steps he talked about 1) Validate 2) Distribute and 3) Measure. After an AMP page ran through the validator one wants to take it live. When this comes up the measuring part is an important one. Some sites need only simple page view tracking, others even want A/B tests. AMP tries to serve them all. There is a `<amp-analytics>` custom element that unifies the analytics work under one hood. It basically serves as a proxy on the client, I assume it has a fixed set of tracking data that it collects and does then spread it to the configured backends (analytics provider). This allows AMP to do the tracking and not every tracker to bring its own code, polute your site and potentially crash it.
+For the A/B testing `<amp-experiment>` can be used. As far as I have understood it's still limited and there might be complex scenarios that it doesn't serve yet.
+
+[rudy]: https://twitter.com/rudygalfi/
+
+## AMP and Ads
+
+As you can see in the [first commit][amp-first-commit] of the AMP project, ads was in there since the beginning. Which is of course not surprising for a company that makes most it's revenue with ads. In current websites ads are often one big slow down factor. It makes sense to tackle this problem. As with the analytics mentioned above AMP also provides strategies to deal with ads.
```

Comment:
`ads was in there` -> `ads were in there`
`most it's revenue` -> `most of it's revenue`
`often one big slow down` -> `often a big slow down`

Reply:
thanks
feel free to commit such text fixes, its more efficient for both of us i think

Reply:
next time ... i am getting this post done currently